snow/networking/benchlist/benchlist.go

```diff
@@ -131,11 +148,22 @@ func (b *queryBenchlist) QueryFailed(validatorID ids.ShortID, requestID uint32)
 		return
 	}
 
+	// Track the message failure and bench [validatorID] if it has
+	// surpassed the threshold
 	key := validatorID.Key()
-	// Add a failure and benches [validatorID] if it has
-	// passed the threshold
-	b.consecutiveFailures[key]++
-	if b.consecutiveFailures[key] >= b.threshold {
+	currentTime := b.clock.Time()
+	failureStreak := b.consecutiveFailures[key]
+	failureStreak.consecutive++
+	if failureStreak.firstFailure.IsZero() {
+		failureStreak.firstFailure = currentTime
+		b.consecutiveFailures[key] = failureStreak
+		// Return early here since a peer will not
+		// be benched after its first failure
+		return
+	}
+	b.consecutiveFailures[key] = failureStreak
+
+	if failureStreak.consecutive >= b.threshold && currentTime.After(failureStreak.firstFailure.Add(b.minimumFailingDuration)) {
```

Comment:
Can we change `.After` to `! .Before` to handle setting `minimumFailingDuration` to 0?